src/QMCWaveFunctions/BsplineFactory/HybridRepCplx.h

```diff
@@ -0,0 +1,218 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2019 QMCPACK developers.
+//
+// File developed by: Ye Luo, yeluo@anl.gov, Argonne National Laboratory
+//
+// File created by: Ye Luo, yeluo@anl.gov, Argonne National Laboratory
+//
+//////////////////////////////////////////////////////////////////////////////////////
+
+
+/** @file HybridRepCplx.h
+ *
+ * hold HybridRepCplx
+ */
+#ifndef QMCPLUSPLUS_HYBRIDREP_CPLX_H
+#define QMCPLUSPLUS_HYBRIDREP_CPLX_H
+
+#include <QMCWaveFunctions/BsplineFactory/HybridRepCenterOrbitals.h>
+namespace qmcplusplus
+{
+/** hybrid representation orbitals combining B-spline orbitals on a grid and atomic centered orbitals.
+ * @tparam SplineBase B-spline orbital class.
+ *
+ * Only works with SplineBase class containing complex splines
+ */
+template<typename SplineBase>
+struct HybridRepCplx : public SplineBase, public HybridRepCenterOrbitals<typename SplineBase::DataType>
+{
+  using HybridBase       = HybridRepCenterOrbitals<typename SplineBase::DataType>;
+  using ST               = typename SplineBase::DataType;
+  using PointType        = typename SplineBase::PointType;
+  using SingleSplineType = typename SplineBase::SingleSplineType;
+  using RealType         = typename SplineBase::RealType;
+  // types for evaluation results
+  using typename SplineBase::GGGVector_t;
+  using typename SplineBase::GradVector_t;
+  using typename SplineBase::HessVector_t;
+  using typename SplineBase::ValueType;
+  using typename SplineBase::ValueVector_t;
+
+  ValueVector_t psi_AO, d2psi_AO;
+  GradVector_t dpsi_AO;
+  Matrix<ST, aligned_allocator<ST>> multi_myV;
+
+  using HybridBase::d2f_dr2;
+  using HybridBase::df_dr;
+  using HybridBase::dist_dr;
+  using HybridBase::dist_r;
+  using SplineBase::myG;
+  using SplineBase::myH;
+  using SplineBase::myL;
+  using SplineBase::myV;
+
+  HybridRepCplx()
+  {
+    this->className = "Hybrid" + this->className;
+    this->KeyWord   = "Hybrid" + this->KeyWord;
+  }
+
+  virtual SPOSet* makeClone() const override { return new HybridRepCplx(*this); }
+
+  inline void resizeStorage(size_t n, size_t nvals)
+  {
+    SplineBase::resizeStorage(n, nvals);
```

Comment:
I don't think splinebase and hybridbase should be inherited from. They make a diamond down to SPOSet which is in general a road to ruin. And conceptually they are elements of the mixed basis. 

One might expect the splinebase and hybridbase to be on somewhat equivalent footing but a splinebase has virtual functions for evaluation and then so you would expect of the hybrid base. However they just have similar/same function names which is certainly problematic enough, yikes.

Reply:
Ok I will look at this more.  I still see a class with a huge accessible namespace available to the methods. I don't see yet why data members needed by the owner can not be accessed from the contained object.

Reply:
Could you explain a bit more? I don't quite understand what you mean in your last reply.

Reply:
I wish I had time to. We can discuss by skype after the Holidays. I don't think it should block the PR.